app/src/main/java/io/halkyon/services/BindApplicationService.java

```diff
@@ -89,27 +91,48 @@ public void createCrossplaneHelmRelease(Service service) throws ClusterConnectEx
 
     private void createSecretForApplication(Claim claim, Credential credential, String url)
             throws ClusterConnectException {
-        String username = credential.username;
-        String password = credential.password;
-
-        if (StringUtils.isNotEmpty(credential.vaultKvPath)) {
-            Map<String, String> vaultSecret = kvSecretEngine.readSecret(credential.vaultKvPath);
-            Set<String> usernames = vaultSecret.keySet();
-            username = usernames.iterator().next();
-            password = vaultSecret.get(username);
-        }
-
         Map<String, String> secretData = new HashMap<>();
         secretData.put(TYPE_KEY, toBase64(claim.type));
         secretData.put(HOST_KEY, toBase64(getHostFromUrl(url)));
         secretData.put(PORT_KEY, toBase64(getPortFromUrl(url)));
         secretData.put(URL_KEY, toBase64(url));
-        secretData.put(USERNAME_KEY, toBase64(username));
-        secretData.put(PASSWORD_KEY, toBase64(password));
 
-        for (CredentialParameter param : credential.params) {
-            secretData.put(param.paramName, toBase64(param.paramValue));
+        String username = "";
+        String password = "";
+        String database = "";
+
+        if (StringUtils.isNotEmpty(credential.username) && StringUtils.isNotEmpty(credential.password)) {
+            username = credential.username;
+            password = credential.password;
+            for (CredentialParameter param : credential.params) {
+                secretData.put(param.paramName, toBase64(param.paramValue));
+            }
+        }
```

Comment:
In a next PR, we will review the code that we have here in order to adapt the logic according to the credential type: simple, vault, etc

We will also define in a new entity, thy keys that we must populate according to the claim binding type (= bindable idea of Ioannis) in order to return the proper fields for a datasource, broker, nosql, etc

Reply:
yes,ok. Can we merge this into the `crossplane-helm`branch then?